marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -535,11 +540,58 @@ function isHubEntityInstance(docUri) {
   return isHubEntityInstance;
 }
 
+function getRecordHistory(docUri) {
+  const history = [];
+  const provResults = xdmp.invokeFunction(function () {
+    const match = {
+      attributes: {
+        location: docUri
+      }
+    };
+    const out = {
+      dateTime: '?',
+      relations: {'associatedWith': '?', 'attributedTo': '?'}
+    };
+    const kvPattern = ps.opTriplePattern(match, out);
+    return op.fromTriples(kvPattern).result();
+  }, { 'database' : xdmp.database(datahub.config.JOBDATABASE)}).toArray();
+
+  if(provResults.length == 0) {
+    const metadata = xdmp.documentGetMetadata(docUri);
```

Comment:
To shorten this function, put all of this logic into a separate function - e.g. "getRecordMetadata".